bindings/bindingtester/bindingtester.py

```diff
@@ -90,7 +90,7 @@ def check_for_errors(self):
             if any([s is not None for s in sequence_nums]):
                 results = {i: r for i, r in results.items() if r.sequence_num(self.specification) == min(sequence_nums)}
             else:
-                results = {i: r for i, r in results.items() if r.matches(min(results.values()), self.specification)}
+                results = {i: r for i, r in results.items() if r.matches_key(min(results.values()), self.specification)}
```

Comment:
Why is this `matches_key` now instead of `matches`?

Reply:
Because this bit is just trying to line up the keys. The actual value matching is done a few lines down.

Reply:
I guess I'm a little unsure about the use of min there to find the minimum result. It's possible that doesn't work in all cases.

Reply:
Hm, I see. Yeah, there's a lot of somewhat non-obvious matching of values here. A lot of the existing code could really use comments. Oh, well.